modules/nw-ingress-gateway-api-deployment-topologies.adoc

```diff
@@ -0,0 +1,135 @@
+== Dedicated gateway example
+An example of a `Gateway` resource, `fin-gateway`:
+
+.Example dedicated `Gateway` resource
+[source,yaml]
+----
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: fin-gateway
+  namespace: openshift-ingress
+spec:
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 8080
+    hostname: "example.com"
+----
+
+An example of the associated `HTTPRoute` resource, `sales-db`, that attaches to the dedicated `Gateway` object:
+
+.Example `HTTPRoute` resource
+[source,yaml]
+----
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: sales-db
+  namespace: openshift-ingress
+spec:
+  parentRefs:
+  - name: fin-gateway
+  rules:
+    - backendRefs:
+        - name: sales-db
+        ¦ port: 8080
+----
+
+The `HTTPRoute` resource must have the `Gateway` object's name as the value for its `parentRefs` field in order to attach to the gateway. Implicitly, the route is assumed to be in the same namespace as the `Gateway` object.
+
+[id="shared-gateway-example_{context}"]
+== Shared gateway example
+An example of a `Gateway` resource, `devops-gateway`, that has a label selector `spec.listeners.allowedRoutes.namespaces` set to match any namespaces containing `shared-gateway-access: "true"`:
+
+.Example shared `Gateway` resource
+[source,yaml]
+----
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: devops-gateway
+  namespace: openshift-ingress
+listeners:
+  - name: https
+    protocol: HTTPS
+    hostname: "example.com"
+    allowedRoutes:
+      namespaces:
+        from: Selector
+        selector:
+        ¦ matchLabels:
+        ¦   shared-gateway-access: "true"
+----
+
+Examples of the allowed namespaces for the `devops-gateway` resource:
+
+.Example `Namespace` resources
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: dev
+  labels:
+    shared-gateway-access: "true"
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ops
+  labels:
+    shared-gateway-access: "true"
+----
+
+An example of two `HTTPRoute` resources, `dev-portal` and `ops-home`, that sit in different namespaces but are attached to the shared gateway:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: HTTPRoute
+metadata:
+  name: dev-portal
+  namespace: dev
+spec:
+  parentRefs:
+  - name: devops-gateway
+    namespace: openshift-ingress
+  rules:
+  - backendRefs:
+    - name: dev-portal
+      port: 8080
+---
+apiVersion: v1
+kind: HTTPRoute
+metadata:
+  name: ops-home
+  namespace: ops
+spec:
+  parentRefs:
+  - name: devops-gateway
+    namespace: openshift-ingress
+  rules:
+  - backendRefs:
+    - name: ops-home
+      port: 8080
+----
+
+With a shared gateway topology, the routes must specify the namespace of the `Gateway` object it wants to attach to. Multiple `Gateway` objects can be deployed and shared across namespaces. When there are multiple shared gateways this becomes conceptually similar to Ingress Controller sharding.
```

Comment:
```suggestion
With a shared gateway topology, the routes must specify the namespace of the `Gateway` object it wants to attach to. Multiple `Gateway` objects can be deployed and shared across namespaces. When there are multiple shared gateways, this becomes conceptually similar to Ingress Controller sharding.
```
It might help to add a noun after "this" in the last sentence. Should it be "When there are multiple shared gateways, this topology becomes conceptually similar to Ingress Controller sharding."?